src/app/[...slug]/page.tsx

```diff
@@ -0,0 +1,32 @@
+import { use } from 'react';
+import { notFound } from 'next/navigation';
+
+import type { AppPage } from '@/lib/types';
+import { getPageByPath } from '@/lib/api.server';
+import { PageComponents } from '@/components/Page/Matcher';
+
+export const revalidate = 10;
+
+export const generateStaticParams = () => {
+  return [];
+};
+
+export const dynamicParams = true;
+
+const CustomPage: AppPage<{ slug: string[] }> = ({ params }) => {
+  const [base, ...rest] = params.slug;
+
+  const path = [`/${base}`].concat(rest).join('/');
```

Comment:
Creo que quedaría más claro de la siguiente forma
```suggestion
  const path = `/${base}/${rest.join('/')}`;
```